app/views/nodes/index.html.slim

```diff
@@ -1,12 +1,65 @@
-h1
-  | Bootstrap cluster
+.alert.alert-info.alert-dismissible role="alert"
+  button.close[type="button" data-dismiss="alert" aria-label="Close"]
+    span[aria-hidden="true"]
+      | &times;
+  i.fa.fa-4x.pull-left aria-hidden="true"
+  span
+    | After choosing the master and clicking "Apply" all other nodes will be set to the minion role
+h1 Nodes
+.row
+  .col-xs-12.discovery-control
+    span #{Minion.count} Nodes found
+    a.btn.btn-default[href="/nodes"] Refresh
+= form_tag(nodes_path, method: "put")
+.nodes-container
+  table.table
+    thead
+      tr
+        th
+          | Id
+        th
+          | Status
+        th
+          | Hostname
+        th
+          | Role
+        th
+          | Highstate
+        th
+          | Master
+    tbody
+      - Minion.all.each do |m|
+        tr
+          th
+            | #{m.id}
+          td
+            - if m.role
+              i.fa.fa-check-circle-o.text-success.fa-2x aria-hidden="true"
+            - else
+              span.fa-stack aria-hidden="true"
+                i.fa.fa-circle.fa-stack-2x.text-success aria-hidden="true"
+                i.fa.fa-refresh.fa-stack-1x.fa-spin.fa-inverse aria-hidden="true"
+          td
+            | #{m.hostname}
+          td
+            | #{m.role}
```

Comment:
In the original design the page where we add nodes is not the same as the monitoring one although this pages serves both roles now. In the add nodes page, what @santiph says, makes sense although @eotchi suggested that we keep this column because in the future we might want to assign roles to minions that already have a role (Federica please confirm this).

Whatever the case, we can create the 2 separate pages (add nodes and monitor) either here or in a separate branch after we merge this one. I suggest we do the latter because this one already includes a lot of cleaning up and it is important to merge it so others can work in parallel on other issues.